### PR TITLE
Clarify course instance cloning instructions

### LINKED ISSUE #1055: Usability issues in the A+ course instance creation (via the gitmanager interface)
The first version of the course creation gitmanager interface in the A+ front was added in #1009.

These issues are based on a small usability study with a teacher. The teacher's goal was to create a new course instance in A+ for the upcoming autumn course.

### Textual hints in the "clone course instance form" (A+ Edit course - Instances tab):

- terminology: "clone course"
  * "Cloning" can be a misleading word. Some may think that it also clones the student data (submissions) (but it does not do that). It would be simpler to just say "create new course instance".
- year and semester/month: teacher thought that the text in the hint "course key in the gitmanager" was confusing. The teacher does not know much about what "gitmanager" or its "course keys" are. The hint could be rephrased without such technical terms.
  * in addition, the teacher thought that if the year and semester are needed to form a unique key across course instances, then why couldn't the A+ course instance URL key be used since it is unique as well? (The user would then not need to insert any "year" or "semester".)
- "update automatically" hint: the text could also emphasize that if the course build fails (in the git manager), then the A+ course will not be broken when you git-push broken course settings by accident. The teacher may be concerned that automatic updates break the system if the teacher makes mistakes in the git course settings.
- "Course instance in Student Information System": the teacher found this confusing. It does not use the word "Sisu" which is more often used at Aalto. It is not clear if this is mandatory and what it is used for. The teacher does not know that this Sisu key is used to fetch course participants from Sisu. The options in the dropdown are a little cryptic; which one should be selected?

### The delay of the first course build

After creating a new course instance, the new instance is still completely empty until the course build has finished in the Git manager. This is not obvious to the teacher and they may get worried at first.
There should be a clear warning about this until the first build has finished.

### Adding the webhook in GitLab

After creating a new course instance, there is no warning that the teacher should manually add the git webhook in GitLab (version.aalto.fi at Aalto). The teacher should also be informed that the webhook URL and secret token are visible in the Edit course - Git manager tab of the new course instance.

### PR DESCRIPTION
**Pull request by ihalaij1**
# Description

**What?**

Clarify the instructions related to course instance cloning based on teacher feedback.

**Why?**

Teacher gave good feedback.

**How?**

See the changes to the code.

Fixes #1055

# Testing

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [x] Manual testing.

Tested that the new info notifications appear.

**Did you test the changes in**

- [x] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

# Translation

- [x] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [x] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [ ] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

### CODE REVIEW

**Comment by ihalaij1** [comment]
Note that this PR automatically removed some unused translations.

**Comment by ihalaij1** [comment]
I did not change the year/semester instructions yet, because we need to first choose what we want to do.
Do we want to remove the year & semester fields, and use the unique URL key as a part of the Git manager key, or do we just change the instructions to be less technical?

If we use the URL keys in Git manager keys, then the Git manager keys will be more random and not as uniform. Currently, the keys for all new courses are in the form of ``COURSE-KEY_YYYYSemester``, but if we use the URL key instead, they will become more random, such as ``COURSE-KEY_2022s``, ``COURSE-KEY_fall2022``, ``COURSE-KEY_2022``.

What do we want?

**Comment by PasiSa** [comment]
> Also, I don't know how we want to explain what the Sisu dropdown selector does, is it mandatory, and how to tell teachers, which one of the cryptic options to choose. If you have suggestions, I will make the changes.

This data comes directly from Sisu: starttime - endtime (type). For each course, Sisu has multiple realizations from near past and near future, and typically the type seems to be "teaching-participation-lectures" for the actual course and "exam-exam" for the exams, but I'm not sure if other values are also possible on some courses. Especially for exams there seems to be multiple entries, because there are multiple exam opportunities during a year. I don't have other ideas at the moment, but to maybe add better description to the description text under the menu to explain how to read the options. The selector is mandatory to get the correct list of students enrolled, because different instances have different students.

**Comment by Mankro** [comment]
> I did not change the year/semester instructions yet, because we need to first choose what we want to do. Do we want to remove the year & semester fields, and use the unique URL key as a part of the Git manager key, or do we just change the instructions to be less technical?
> 
> If we use the URL keys in Git manager keys, then the Git manager keys will be more random and not as uniform. Currently, the keys for all new courses are in the form of `COURSE-KEY_YYYYSemester`, but if we use the URL key instead, they will become more random, such as `COURSE-KEY_2022s`, `COURSE-KEY_fall2022`, `COURSE-KEY_2022`.
> 
> What do we want?

I think we could keep the current naming scheme `COURSE-CODE_YYYYSemester` and then we wouldn't remove the form fields for the year and semester. There is some value in keeping the internal gitmanager course keys more structured and organized. (Though, if the teacher can still actually change the course key in the Edit course - Git manager tab, then they can still override the keys with nonsense.)

Even though one teacher didn't like entering extra info like the year and semester, I still think it is not too much to ask. The teacher should know on which year and semester they are running their courses.

**Comment by Mankro** [comment]
Upstream master has changed after your other PR was merged (#1058). You may rebase and fix django.po translations.

**Comment by PasiSa** [comment]
> > Also, I don't know how we want to explain what the Sisu dropdown selector does, is it mandatory, and how to tell teachers, which one of the cryptic options to choose. If you have suggestions, I will make the changes.
> 
> This data comes directly from Sisu: starttime - endtime (type). For each course, Sisu has multiple realizations from near past and near future, and typically the type seems to be "teaching-participation-lectures" for the actual course and "exam-exam" for the exams, but I'm not sure if other values are also possible on some courses. Especially for exams there seems to be multiple entries, because there are multiple exam opportunities during a year. I don't have other ideas at the moment, but to maybe add better description to the description text under the menu to explain how to read the options. The selector is mandatory to get the correct list of students enrolled, because different instances have different students.

How about the following description text to explain the dropdown menu?
"Course instance from which students are enrolled. It shows the start and ending times of the course instance, along with the enrollment type (e.g. course participations or exam enrollment)"

The problem is that this menu is built by the organization-specific SIS plugin. If some other university started using this feature, they might have completely different format in menu entries.

**Comment by Mankro** [comment]
> How about the following description text to explain the dropdown menu? "Course instance from which students are enrolled. It shows the start and ending times of the course instance, along with the enrollment type (e.g. course participation or exam enrollment)"
> 
> The problem is that this menu is built by the organization-specific SIS plugin. If some other university started using this feature, they might have completely different format in menu entries.

This sounds good to me! Let's return to the "other universities" later in the future if this becomes a problem.

**Comment by ihalaij1** [comment]
I made all the changes now, this is ready for a review.

File: locale/en/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -2138,11 +2140,15 @@ msgstr ""
...
-msgstr "Course instance in Student Information System"
+msgstr "Course instance in Student Information System (SIS)"
 
 #: edit_course/course_forms.py
 msgid "LABEL_SIS_INSTANCE_HELPTEXT"
-msgstr "Select the course instance in the Student Information System."
+msgstr ""
+"Course instance from which students are enrolled. It shows the start and "
+"ending times of the course instance, along with the enrollment type (e.g. "
+"course participation or exam enrollment). The below option only applies if "
```
Comment (Mankro) [inline locale/en/LC_MESSAGES/django.po:2150]:
```suggestion
"ending times of the course instance, along with the enrolment type (e.g., "
"course participation or exam enrolment). The option below only applies if "
```
We have used the spelling "enrolment" in the UI.

**Review by Mankro — CHANGES_REQUESTED** [review]
Great! I thought of a few small changes.

**Comment by ihalaij1** [comment]
These changes have been implemented now!

**Review by Mankro — APPROVED** [review]
Excellent! I will merge this now.

**Pull request merged**